pkg/build/gobuild.go

```diff
@@ -968,8 +968,12 @@ func (g *gobuild) buildAll(ctx context.Context, ref string, baseIndex v1.ImageIn
 		if err != nil {
 			return nil, fmt.Errorf("error getting matching image from index: %w", err)
 		}
-		// Carry forward the base index's annotations, which include base image annotations.
```

Comment:
I think I'm okay with it, but just to make sure, this change also has the result of _not_ carrying over any base image annotations, and only settings these new annotations.

I think that's probably fine, I just want to make sure this was a conscious decision.

Reply:
I got the impression that the main use case of this was to get the base image annotation carried forward.  Without another use case in mind it felt like apples/oranges to me, where blindly copying context could lead to confusion.  I haven't had enough ☕ to produce contrived examples yet

Reply:
Yeah, having had two ☕ s so far I'm only becoming more convinced that blindly carrying forward the annotations is wrong. These may include things like source repo or description, that don't make sense for derived images.

Reply:
Great examples, and not even contrived! 😂